### ISSUE TITLE
Compiling issue for 32-bit

### ISSUE THREAD

**Issue by bqminh** [upsj/terraphast-one#16]
When compiling terraphast on 32-bit Windows with Clang, I got the following errors. Is there a plan to support 32-bit? That'd be good so that it can be integrated into 32-bit application.

Thanks, Minh

```
In file included from \Users\minh\Dropbox\iqtree-git\terraphast\lib\advanced.cpp:8:
In file included from /Users/minh/Dropbox/iqtree-git/terraphast/lib/multitree_iterator.hpp:5:
In file included from /Users/minh/Dropbox/iqtree-git/terraphast/lib/small_bipartition.hpp:10:
/Users/minh/Dropbox/iqtree-git/terraphast/lib/bits.hpp:15:46: error: implicit conversion loses integer precision:
      'unsigned long long' to 'terraces::index' (aka 'unsigned int') [-Werror,-Wshorten-64-to-32]
inline index set_mask(index i) { return 1ull << (i & (word_bits - 1)); }
                                 ~~~~~~ ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
In file included from \Users\minh\Dropbox\iqtree-git\terraphast\lib\advanced.cpp:8:
In file included from /Users/minh/Dropbox/iqtree-git/terraphast/lib/multitree_iterator.hpp:5:
/Users/minh/Dropbox/iqtree-git/terraphast/lib/small_bipartition.hpp:37:34: error: implicit conversion loses integer
      precision: 'unsigned long long' to 'terraces::index' (aka 'unsigned int') [-Werror,-Wshorten-64-to-32]
        void reset() { m_cur_bip = 1ull << bitscan(m_mask); }
                                 ~ ~~~~~^~~~~~~~~~~~~~~~~~
/Users/minh/Dropbox/iqtree-git/terraphast/lib/small_bipartition.hpp:48:11: error: non-constant-expression cannot be
      narrowed from type 'unsigned long long' to 'terraces::index' (aka 'unsigned int') in initializer list
      [-Wc++11-narrowing]
                return {(1ull << num_leaves) - 1};
                        ^~~~~~~~~~~~~~~~~~~~~~~~
/Users/minh/Dropbox/iqtree-git/terraphast/lib/small_bipartition.hpp:48:11: note: insert an explicit cast to silence this

      issue
                return {(1ull << num_leaves) - 1};
                        ^~~~~~~~~~~~~~~~~~~~~~~~
                        static_cast<index>(     )
/Users/minh/Dropbox/iqtree-git/terraphast/lib/small_bipartition.hpp:48:32: error: implicit conversion loses integer
      precision: 'unsigned long long' to 'terraces::index' (aka 'unsigned int') [-Werror,-Wshorten-64-to-32]
                return {(1ull << num_leaves) - 1};
                       ~~~~~~~~~~~~~~~~~~~~~~^~~
4 errors generated.
```

**Comment by upsj** [comment]
This is probably fixed - the whole code should be word size-agnostic now - but I could not yet test all build configurations, so best wait with the integration until I've built or had access to a 32bit test environment.

**Comment by bqminh** [comment]
That's cool thanks! You simply need to add option -m32 into gcc/clang compiler. E.g. setting CMAKE_C_FLAGS and CMAKE_CXX_FLAGS ahead of the build. This will compile 32-bit binary, even if you are running on 64-bit machine.

**Comment by upsj** [comment]
Thanks! There seem to be some edge case issues with the bit operations in `multitree_iterator` with gcc, but not with cl (Microsoft VC++), other than that, everything works.

**Comment by upsj** [comment]
This should be fixed by 39620018
